app/helpers/hourglass/chart_helper.rb

```diff
@@ -37,7 +37,13 @@ def hours_per_date(query)
       query.total_by_group_for(:hours).transform_values do |totals_by_column|
         totals_by_column = {default: totals_by_column} unless query.main_query.group_by_statement
         totals_by_column.transform_keys! { |_| :default } if query.main_query.group_by == 'date'
-        Hash[totals_by_column.map { |column, total| [column, time_booking_total(total)] }]
+        Hash[totals_by_column.map { |column, total| [column, unrounded_total(total)] }]
+      end
+    end
+
+    def unrounded_total(total)
+      total.reduce(0.0) do |sum, total_by_project|
+        sum + total_by_project.sum.to_f.round(2)
```

Comment:
I can't test this right now, but I am pretty sure, that this doesn't work. Examining the original code (https://github.com/hicknhack-software/redmine_hourglass/blob/9be80fad9664cf8f4ed6c4db8ac52db7f3c8543a/app/helpers/hourglass/list_helper.rb#L98) it suggests that `total_by_project` is an array (or maybe an hash, not sure), so summing there makes no sense.

Please double check this.

Reply:
I don't understand your objection, this is exactly why the array is folded with with `reduce`. Actually, it is basically the same code as `time_booking_total`

Reply:
Alright, seems like i should have put the comment on the correct line.
I am talking about line in the reduce block.